### ISSUE TITLE
ForegroundServiceDidNotStartInTimeException ^0.6.4

### ISSUE THREAD

**Issue by miguelflores1993** [aws-amplify/amplify-flutter#1935]
### Description

E/AndroidRuntime(26016): FATAL EXCEPTION: main
E/AndroidRuntime(26016): Process: com.deltaxlat.appdriver.dev, PID: 26016
E/AndroidRuntime(26016): android.app.ForegroundServiceDidNotStartInTimeException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{ceca0f1 u0 com.deltaxlat.appdriver.dev/com.amazonaws.mobileconnectors.s3.transferutility.TransferService}
E/AndroidRuntime(26016): 	at android.app.ActivityThread.throwRemoteServiceException(ActivityThread.java:2027)
E/AndroidRuntime(26016): 	at android.app.ActivityThread.access$2900(ActivityThread.java:273)
E/AndroidRuntime(26016): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2252)
E/AndroidRuntime(26016): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime(26016): 	at android.os.Looper.loopOnce(Looper.java:210)
E/AndroidRuntime(26016): 	at android.os.Looper.loop(Looper.java:299)
E/AndroidRuntime(26016): 	at android.app.ActivityThread.main(ActivityThread.java:8256)

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [X] Storage

### Steps to Reproduce

E/AndroidRuntime(26016): FATAL EXCEPTION: main
E/AndroidRuntime(26016): Process: com.deltaxlat.appdriver.dev, PID: 26016
E/AndroidRuntime(26016): android.app.ForegroundServiceDidNotStartInTimeException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{ceca0f1 u0 com.deltaxlat.appdriver.dev/com.amazonaws.mobileconnectors.s3.transferutility.TransferService}
E/AndroidRuntime(26016): 	at android.app.ActivityThread.throwRemoteServiceException(ActivityThread.java:2027)
E/AndroidRuntime(26016): 	at android.app.ActivityThread.access$2900(ActivityThread.java:273)
E/AndroidRuntime(26016): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2252)
E/AndroidRuntime(26016): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime(26016): 	at android.os.Looper.loopOnce(Looper.java:210)
E/AndroidRuntime(26016): 	at android.os.Looper.loop(Looper.java:299)
E/AndroidRuntime(26016): 	at android.app.ActivityThread.main(ActivityThread.java:8256)

### Platforms

- [ ] iOS
- [X] Android

### Environment

```bash
[√] Flutter (Channel stable, 3.0.5, on Microsoft Windows [Versi¢n 10.0.22000.795], locale es-BO)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[√] Chrome - develop for the web
[√] Android Studio (version 2021.1)
[√] VS Code (version 1.69.2)
[√] Connected device (3 available)
[!] HTTP Host Availability
    X HTTP host "https://cloud.google.com/" is not reachable. Reason: Failed to connect to host in 10 seconds
```

### Dependencies

```bash
- amplify_auth_cognito 0.6.4 [amplify_auth_cognito_android amplify_auth_cognito_ios amplify_core aws_common collection flutter meta plugin_platform_interface]
- amplify_flutter 0.6.4 [amplify_core amplify_datastore_plugin_interface amplify_flutter_android amplify_flutter_ios aws_common collection flutter meta plugin_platform_interface]
- amplify_storage_s3 0.6.4 [amplify_storage_s3_android amplify_storage_s3_ios amplify_core aws_common flutter meta plugin_platform_interface]
- background_fetch 1.1.0 [flutter]
- bloc_concurrency 0.2.0 [bloc stream_transform]
- cached_network_image 3.2.1 [flutter flutter_cache_manager octo_image cached_network_image_platform_interface cached_network_image_web]
- camera 0.10.0 [camera_android camera_avfoundation camera_platform_interface camera_web flutter flutter_plugin_android_lifecycle quiver]
- collection 1.16.0
- connectivity_plus 2.3.6 [flutter connectivity_plus_platform_interface connectivity_plus_linux connectivity_plus_macos connectivity_plus_web connectivity_plus_windows]
- cron 0.5.0 [clock]
- cupertino_icons 1.0.5
- dio 4.0.6 [http_parser path]
- double_back_to_close_app 2.1.0 [flutter]
- easy_mask 2.0.1 [flutter]
- equatable 2.0.3 [collection meta]
- file_picker 5.0.0 [flutter flutter_web_plugins flutter_plugin_android_lifecycle plugin_platform_interface ffi path win32]
- firebase_analytics 9.3.0 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 1.20.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_dynamic_links 4.3.3 [firebase_core firebase_core_platform_interface firebase_dynamic_links_platform_interface flutter meta plugin_platform_interface]
- firebase_messaging 12.0.1 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_performance 0.8.2+1 [firebase_core firebase_core_platform_interface firebase_performance_platform_interface firebase_performance_web flutter]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_background_geolocation 4.7.0 [flutter collection background_fetch]
- flutter_bloc 8.0.1 [flutter bloc provider]
- flutter_dotenv 5.0.2 [flutter]
- flutter_image_compress 1.1.0 [flutter]
- flutter_local_notifications 9.7.0 [clock flutter flutter_local_notifications_linux flutter_local_notifications_platform_interface timezone]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_polyline_points 1.0.0 [flutter http]
- flutter_rating_bar 4.0.1 [flutter]
- font_awesome_flutter 10.1.0 [flutter]
- geocoding 2.0.4 [flutter geocoding_platform_interface]
- get_it 7.2.0 [async collection]
- google_fonts 3.0.1 [flutter http path_provider crypto]
- google_maps_flutter 2.1.10 [flutter flutter_plugin_android_lifecycle google_maps_flutter_platform_interface]
- hive 2.2.3 [meta crypto]
- hydrated_bloc 8.1.0 [bloc hive meta synchronized]
- image_picker 0.8.5+3 [flutter image_picker_android image_picker_for_web image_picker_ios image_picker_platform_interface]
- in_app_update 3.0.0 [flutter]
- intl 0.17.0 [clock path]
- mime 1.0.2
- multi_select_flutter 4.1.2 [flutter collection]
- package_info_plus 1.4.3 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- path 1.8.1
- path_provider 2.0.11 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- percent_indicator 4.2.2 [flutter]
- permission_handler 9.2.0 [flutter meta permission_handler_android permission_handler_apple permission_handler_windows permission_handler_platform_interface]
- pin_code_fields 7.4.0 [flutter]
- pull_to_refresh 1.0.0 [flutter]
- pusher_client 2.0.0 [flutter]
- random_string 2.3.1
- sentry_flutter 6.8.1 [flutter flutter_web_plugins sentry package_info_plus meta]
- shared_preferences 2.0.15 [flutter shared_preferences_android shared_preferences_ios shared_preferences_linux shared_preferences_macos shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- sizer 2.0.15 [universal_io flutter]
- smooth_page_indicator 1.0.0+2 [flutter]
- timeago 3.2.2
- url_launcher 6.1.5 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- vector_math 2.1.2
- webview_flutter 3.0.4 [flutter webview_flutter_android webview_flutter_platform_interface webview_flutter_wkwebview]

transitive dependencies:
- amplify_auth_cognito_android 0.6.4 [flutter]
- amplify_auth_cognito_ios 0.6.4 [amplify_core flutter]
- amplify_core 0.6.4 [aws_common collection flutter intl json_annotation meta plugin_platform_interface uuid]
- amplify_datastore_plugin_interface 0.6.4 [amplify_core collection flutter meta]
- amplify_flutter_android 0.6.4 [flutter]
- amplify_flutter_ios 0.6.4 [amplify_core flutter]
- amplify_storage_s3_android 0.6.4 [flutter]
- amplify_storage_s3_ios 0.6.4 [flutter]
- args 2.3.1
- async 2.8.2 [collection meta]
- aws_common 0.1.1 [async collection http meta stream_transform uuid]
- bloc 8.0.3 [meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- cached_network_image_platform_interface 1.0.0 [flutter flutter_cache_manager]
- cached_network_image_web 1.0.1 [flutter flutter_cache_manager cached_network_image_platform_interface]
- camera_android 0.10.0 [camera_platform_interface flutter flutter_plugin_android_lifecycle stream_transform]
- camera_avfoundation 0.9.8+2 [camera_platform_interface flutter stream_transform]
- camera_platform_interface 2.2.0 [cross_file flutter plugin_platform_interface stream_transform]
- camera_web 0.3.0 [camera_platform_interface flutter flutter_web_plugins stream_transform]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- connectivity_plus_linux 1.3.1 [flutter connectivity_plus_platform_interface meta nm]
- connectivity_plus_macos 1.2.4 [connectivity_plus_platform_interface flutter]
- connectivity_plus_platform_interface 1.2.1 [flutter meta plugin_platform_interface]
- connectivity_plus_web 1.2.3 [connectivity_plus_platform_interface flutter_web_plugins flutter]
- connectivity_plus_windows 1.2.2 [connectivity_plus_platform_interface flutter]
- cross_file 0.3.3+1 [js meta]
- crypto 3.0.2 [typed_data]
- dbus 0.7.7 [args ffi meta xml]
- fake_async 1.3.0 [clock collection]
- ffi 2.0.1
- file 6.1.2 [meta path]
- firebase 9.0.3 [http http_parser js]
- firebase_analytics_platform_interface 3.3.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.4.2 [firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_core_platform_interface 4.5.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 1.7.1 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_dynamic_links_platform_interface 0.2.3+8 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_platform_interface 4.1.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 3.1.0 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- firebase_performance_platform_interface 0.1.1+12 [firebase_core flutter plugin_platform_interface]
- firebase_performance_web 0.1.1+1 [firebase firebase_core firebase_core_web firebase_performance_platform_interface flutter flutter_web_plugins js]
- flutter_blurhash 0.7.0 [flutter]
- flutter_cache_manager 3.3.0 [clock collection file flutter http path path_provider pedantic rxdart sqflite uuid]
- flutter_local_notifications_linux 0.5.0+1 [flutter flutter_local_notifications_platform_interface dbus path xdg_directories]
- flutter_local_notifications_platform_interface 5.0.0 [flutter plugin_platform_interface]
- flutter_plugin_android_lifecycle 2.0.7 [flutter]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- geocoding_platform_interface 2.0.1 [flutter meta plugin_platform_interface]
- google_maps_flutter_platform_interface 2.2.1 [collection flutter plugin_platform_interface stream_transform]
- http 0.13.4 [async http_parser meta path]
- http_parser 4.0.1 [collection source_span string_scanner typed_data]
- image_picker_android 0.8.5+1 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface]
- image_picker_for_web 2.1.8 [flutter flutter_web_plugins image_picker_platform_interface]
- image_picker_ios 0.8.5+6 [flutter image_picker_platform_interface]
- image_picker_platform_interface 2.5.0 [cross_file flutter http plugin_platform_interface]
- js 0.6.4
- json_annotation 4.6.0 [meta]
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- nested 1.0.0 [flutter]
- nm 0.5.0 [dbus]
- octo_image 1.0.2 [flutter flutter_blurhash]
- package_info_plus_linux 1.0.5 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.3.0 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.5 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 2.0.0 [package_info_plus_platform_interface ffi flutter win32]
- path_provider_android 2.0.16 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.10 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.4 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.0 [ffi flutter path path_provider_platform_interface win32]
- pedantic 1.11.1
- permission_handler_android 9.0.2+1 [flutter permission_handler_platform_interface]
- permission_handler_apple 9.0.4 [flutter permission_handler_platform_interface]
- permission_handler_platform_interface 3.7.0 [flutter meta plugin_platform_interface]
- permission_handler_windows 0.1.0 [flutter permission_handler_platform_interface]
- petitparser 5.0.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- process 4.2.4 [file path platform]
- provider 6.0.3 [collection flutter nested]
- quiver 3.1.0 [matcher]
- rxdart 0.27.5
- sentry 6.8.1 [http meta stack_trace uuid]
- shared_preferences_android 2.0.12 [flutter shared_preferences_platform_interface]
- shared_preferences_ios 2.1.1 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.1.1 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_macos 2.0.4 [flutter shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.0.0 [flutter]
- shared_preferences_web 2.0.4 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.1.1 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sky_engine 0.0.99
- source_span 1.8.2 [collection path term_glyph]
- sqflite 2.0.3 [flutter sqflite_common path]
- sqflite_common 2.2.1+1 [synchronized path meta]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- stream_transform 2.0.0
- string_scanner 1.1.0 [charcode source_span]
- synchronized 3.0.0+2
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- timezone 0.8.0 [path]
- typed_data 1.3.1 [collection]
- universal_io 2.0.4 [collection crypto meta typed_data]
- url_launcher_android 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.0 [flutter plugin_platform_interface]
- url_launcher_web 2.0.12 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- uuid 3.0.6 [crypto]
- webview_flutter_android 2.9.2 [flutter webview_flutter_platform_interface]
- webview_flutter_platform_interface 1.9.1 [flutter meta plugin_platform_interface]
- webview_flutter_wkwebview 2.9.1 [flutter path webview_flutter_platform_interface]
- win32 2.7.0 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- xml 6.1.0 [collection meta petitparser]
```

### Device

XIAOMI MI 11T

### OS

ANDROID 12

### CLI Version

8.0.0

**Comment by dnys1** [comment]
Hi @miguelflores1993 - we are aware of the issue and will have a fix out shortly. Thank you for the detailed report and apologies for any inconvenience.

**Comment by dnys1** [comment]
Hi @miguelflores1993 - we've just published v0.6.5. Can you please verify this is fixed?

**Comment by dnys1** [comment]
Closing this for now as a fix was released in v0.6.5. Please feel free to re-open if you continue experience issues.
